### PR TITLE
Add bufsize and maxrate ffmpeg rendering options

### PR DESCRIPTION
**Pull request by llccrr**
#3232 

- [x] Link issues that the PR is affecting
- [x]  Global workflow working
- [x] Add a unit test to verify the integrity of function params
- [ ] Enrich documentation

### CODE REVIEW

File: packages/cli/src/config/bitrate.ts [hunk 1]
```diff
@@ -17,3 +17,27 @@ export const setVideoBitrate = (bitrate: string | null) => {
...
+}
+
+export const getVideoMaxRate = () => {
+	return maxRate;
+}
+
+/**
+ * bufSize is not a bitrate, but it is a bitrate-related option and get validated like a bitrate.
+ */
+let bufSize: string | null = null;
```
Comment (llccrr) [inline packages/cli/src/config/bitrate.ts]: Should I let bufSize setter and getter in the bitrate.ts for "simplicity" or should I create a new file ?

File: packages/renderer/src/crf.ts [hunk 2]
```diff
@@ -71,17 +71,28 @@ export const validateQualitySettings = ({
...
+	maxRate: string | null | undefined;
+	bufSize: string | null | undefined;
 }): string[] => {
 	if (crf && videoBitrate) {
 		throw new Error(
 			'"crf" and "videoBitrate" can not both be set. Choose one of either.',
 		);
 	}
 
+	const bufSizeArray = bufSize ? ['-bufsize', bufSize] : [];
```
Comment (llccrr) [inline packages/renderer/src/crf.ts]: I think bufSize and maxRate are compatible with crf or videobitrate, so I am just spreading in both of them

Comment (llccrr) [reply packages/renderer/src/crf.ts:89]: Maybe I missed an error to handle here ?

File: packages/cli/src/preview-server/routes/add-render.ts [hunk 3]
```diff
@@ -41,6 +41,11 @@ export const handleAddRender: ApiHandler<AddRenderRequest, undefined> = ({
 				pixelFormat: input.pixelFormat,
 				audioBitrate: input.audioBitrate,
 				videoBitrate: input.videoBitrate,
+				// TODO : Remove ts-expect-error
```
Comment (llccrr) [inline packages/cli/src/preview-server/routes/add-render.ts:44]: Might need help on this one..

File: packages/cloudrun/src/api/render-media-on-cloudrun.ts [hunk 4]
```diff
@@ -175,6 +181,8 @@ const internalRenderMediaOnCloudrunRaw = async ({
 		audioCodec: audioCodec ?? null,
 		audioBitrate: audioBitrate ?? null,
 		videoBitrate: videoBitrate ?? null,
+		bufSize: maxRate ?? null,
+		maxRate: bufSize ?? null,
```
Comment (JonnyBurger) [inline packages/cloudrun/src/api/render-media-on-cloudrun.ts]: Caught it! 😛

Comment (llccrr) [reply packages/cloudrun/src/api/render-media-on-cloudrun.ts]: Wow, well done ! Ahah

**Pull request merged**